src/action/memory_size.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020 Google LLC
+//
+// Use of this source code is governed by an MIT-style license that can be found
+// in the LICENSE file or at https://opensource.org/licenses/MIT.
+
+//! A handler and associated types for the `GetMemorySize` action.
```

Comment:
I would prefer to avoid using the `GetMemorySize` identifier, they are quite ugly and kind of a protocol detail. Just write "the memory size action" everywhere (except for where you need to register it and the pull request name).